db/queries/categories.js

```diff
@@ -0,0 +1,61 @@
+const db = require('../connection');
+/* Fetch Categories and count of all items in that particular*/
+
+const getCategory = (categoryId) => {
+  const query = `SELECT categories.*, COUNT(items.id) AS total_items
+  FROM categories
+  JOIN items
+  ON categories.id = categories_id
+  WHERE categories.id = $1
+  GROUP BY categories.id;`;
+  return db.query(query,[categoryId])
+    .then(data => {
+      return data.rows[0];
+    })
+    .catch((err) => {
+      console.log(err.message);
+    });
+};
+
+/* Adding a new category */
+
+const addCategory = (categoryId, owner_id, categoryName) => {
+  const query = `INSERT INTO categories(owner_id, name, created_at)
+  VALUES ($2, $3, now()) RETURNING id;`
+  return db.query(query, [categoryId, owner_id, categoryName])
+  .then(data => {
+    return data.rows[0];
+  })
+  .catch((err) => {
+    console.log(err.message);
+  });
+}
+
+const deleteCategory = (categoryId) => {
+  const query = `DELETE FROM categories WHERE id = $1;`
+  return db.query(query, [categoryId])
+  .catch((err) => {
+    console.log(err.message);
+  });
+}
+
+const updateCategory = (categoryId, name) => {
+  const query = `UPDATE categories
+  SET name = $1
+  WHERE id = $2;`
+  return db.query(query,[name, categoryId])
+    .then(data => {
```

Comment:
This SQL query does not have a RETURNING statement, so I'm not sure if any data would be returned.